### PR TITLE
ci(sca): run datadog sca in the ci

### PR DESCRIPTION
**Pull request by juliendoutre**
### What does this PR do?
Add a new Gitlab CI job listing this repository dependencies with trivy and uploading them to Datadog. This will make it possible to use the SCA product for this repository (see https://app.datadoghq.com/ci/code-analysis?limit=100&offset=0).

### Motivation
@DataDog/software-integrity-and-trust partners with @DataDog/static-analysis to dogfood their SCA product and secure Datadog's supply chain.

### Additional Notes
[Successful CI run](https://gitlab.ddbuild.io/DataDog/apm-reliability/dd-trace-py/-/jobs/440743128)

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ stages:
 include:
   - remote: https://gitlab-templates.ddbuild.io/apm/packaging.yml
   - local: ".gitlab/benchmarks.yml"
+  - /.gitlab/software_composition_analysis.yaml
```
Comment (P403n1x87) [inline .gitlab-ci.yml:10]:
```suggestion
  - local: .gitlab/software_composition_analysis.yaml
```

**Review by P403n1x87 — APPROVED** [review]
Nit, otherwise LGTM, thanks!

**Comment by majorgreys** [comment]
@juliendoutre Do we need this to be run on PR branches? This job would not be run on PRs from external contributors. Would it be enough to run it on main once PRs have gotten merged?

**Comment by juliendoutre** [comment]
> @juliendoutre Do we need this to be run on PR branches? This job would not be run on PRs from external contributors. Would it be enough to run it on main once PRs have gotten merged?

@majorgreys yes definitely! We're interested in starting dogfooding this product and running it only on the `main` branch is already a good first step for us!

**Comment by juliendoutre** [comment]
Closing this PR as SCA is working on a way to enable their product on all repos of an organization at once.

**Pull request closed without merging**